phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
I clearly prefer rxu's proposal.  Much cleaner, and better BC (perfect, I should say).
But just rename `not_enableable_reason` to `get_errors` and voila!  That's it.
The fact that $errs is in the class is an implementation detail, that might be the norm, but should not be imposed or required.